diff --git a/node_modules/@u4/opencv4nodejs/binding.gyp b/node_modules/@u4/opencv4nodejs/binding.gyp
index 5e87e2f..51d0d24 100644
--- a/node_modules/@u4/opencv4nodejs/binding.gyp
+++ b/node_modules/@u4/opencv4nodejs/binding.gyp
@@ -163,7 +163,8 @@
 	          {
 	            "link_settings": {
 	              "libraries": [
-					"-Wl,-rpath,@loader_path/../../../opencv-build/opencv/build/lib"
+					"-Wl,-rpath,@loader_path/../../../opencv-build/opencv/build/lib",
+          "-Wl,-rpath,@loader_path/../../../opencv-build/dist/opencv-4.5.4-8b1ea/build/lib"
 	              ],
 	            }
 	          }
